src/file_organizer/api/routers/setup.py

```
@@ -171,3 +173,50 @@ def complete_setup(
+
+    Returns:
+        BrowseFolderResponse with the selected absolute path, or an indication
+        that a native dialog is not available in this environment.
+    """
+    if sys.platform != "darwin":
+        # Linux (Docker) and Windows cannot show a macOS Finder dialog.
+        return BrowseFolderResponse(path="", available=False)
+
+    try:
+        result = subprocess.run(
+            ["osascript", "-e", "POSIX path of (choose folder)"],
```

Ruff (0.15.9)

[error] Starting a process with a partial executable path

(S607)